packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -537,12 +551,37 @@ class RadialGradient extends Gradient {
   /// ![](https://flutter.github.io/assets-for-api-docs/assets/dart-ui/tile_mode_repeated_radial.png)
   final TileMode tileMode;
 
+  /// The focal point of the gradient.  If specified, the gradient will appear
+  /// to be focused along the vector from [center] to focal. 
+  /// 
+  /// See [center] for a description of how the coordinates are mapped.
+  /// 
+  /// If this value is specified and [focalRadius] > 0.0, care should be taken
+  /// to ensure that either this value or [center] will not both resolve to
+  /// [Offset.zero], which would fail to create a valid gradient.
+  final AlignmentGeometry focal;
+
+  /// The radius of the focal point of gradient, as a fraction of the shortest
+  /// side of the paint box.
+  ///
+  /// For example, if a radial gradient is painted on a box that is
+  /// 100.0 pixels wide and 200.0 pixels tall, then a radius of 1.0
+  /// will place the 1.0 stop at 100.0 pixels from the [focus].
+  /// 
+  /// If this value is specified and is greater than 0.0, either [focal] or
+  /// [center] must not resolve to [Offset.zero], which would fail to create
+  /// a valid gradient.
+  final double focalRadius;
+
   @override
   Shader createShader(Rect rect, { TextDirection textDirection }) {
     return new ui.Gradient.radial(
       center.resolve(textDirection).withinRect(rect),
       radius * rect.shortestSide,
       colors, _impliedStops(), tileMode,
+      null, // transform
+      focal?.resolve(textDirection)?.withinRect(rect),
```

Comment:
To avoid the redundant null check, consider:

`focal == null ? null : focal.resolve(textDirection).withinRect(rect),`